web/packages/teleport/src/services/user/user.ts

```diff
@@ -121,10 +121,26 @@ const service = {
     await session.renewSession({ reloadUser: true }, signal);
   },
 
-  checkUserHasAccessToRegisteredResource(): Promise<boolean> {
-    return api
-      .get(cfg.getCheckAccessToRegisteredResourceUrl())
-      .then(res => Boolean(res.hasResource));
+  async checkUserHasAccessToAnyRegisteredResource() {
+    const clusterId = cfg.proxyCluster;
+
+    const res = await api
+      .get(
+        cfg.getUnifiedResourcesUrl(clusterId, {
+          limit: 1,
+          sort: {
+            fieldName: 'name',
+            dir: 'ASC',
+          },
+        })
+      )
+      .catch(err => {
+        // eslint-disable-next-line no-console
+        console.error('Error checking access to registered resources', err);
+        return { items: [] };
+      });
+
+    return !!res?.items?.length;
```

Comment:
Won't this also return true for data like `resp = {items: [null/undefined]}` ? I guess the backend wont return null here but just thinking out loud.

Reply:
if you mean `null`/`undefined` as entries in the list, in that case it would return true. could do something like `return !!res?.items?.some((i) => !!i)` but that feels a bit overkill to me...

Reply:
Yes, since Go's json marshaller can send `[null]` array if there are zero items in the slice and the field is not initialized with `make()`. Though we are good here since ListUnifiedResources does not send such data. Your call how safe we want to be here ;)